From b33a6b2a178082567c8f6dead402676d2d5b8101 Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Tue, 11 Feb 2020 13:07:19 +0530
Subject: [PATCH 1132/1851] dmaengine: xilinx: axidmatest: First enable receive
 path

commit 27a6c429f163aad58f28f821b37c589e13406b4e from
https://github.com/Xilinx/linux-xlnx.git

After dma configuration, first start receive and then transmit data mover.
It avoid error scenarios which could be triggered due to the delay between
transmit and receive dma start. Fixes packet drop with MCDMA dmaengine.

Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: not-upstreamable
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/dma/xilinx/axidmatest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/axidmatest.c b/drivers/dma/xilinx/axidmatest.c
index 4d69dc41c6d7..63ef98510712 100644
--- a/drivers/dma/xilinx/axidmatest.c
+++ b/drivers/dma/xilinx/axidmatest.c
@@ -401,8 +401,8 @@ static int dmatest_slave_func(void *data)
 			failed_tests++;
 			continue;
 		}
-		dma_async_issue_pending(tx_chan);
 		dma_async_issue_pending(rx_chan);
+		dma_async_issue_pending(tx_chan);
 
 		tx_tmo = wait_for_completion_timeout(&tx_cmp, tx_tmo);
 
-- 
2.31.1

